In [15]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import matplotlib.pyplot as plt

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/netflix-shows/netflix_titles.csv


Some basic analysis and cleaning of the "netflix-shows" dataset.

In [16]:
df = pd.read_csv("/kaggle/input/netflix-shows/netflix_titles.csv")
df.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


In [18]:
df.shape

(8807, 12)

**Dropping all the rows with atleast one null value in them.**

In [19]:
df = df.dropna()

**Dropping the columns which I don't find useful.**

In [20]:
df = df.drop(columns = ["show_id", "cast", "description"])

In [21]:
df.shape

(5332, 9)

In [22]:
df.head()

type                          title             director  \
7     Movie                        Sankofa         Haile Gerima   
8   TV Show  The Great British Baking Show      Andy Devonshire   
9     Movie                   The Starling       Theodore Melfi   
12    Movie                   Je Suis Karl  Christian Schwochow   
24    Movie                          Jeans           S. Shankar   

                                              country          date_added  \
7   United States, Ghana, Burkina Faso, United Kin...  September 24, 2021   
8                                      United Kingdom  September 24, 2021   
9                                       United States  September 24, 2021   
12                            Germany, Czech Republic  September 23, 2021   
24                                              India  September 21, 2021   

    release_year rating   duration  \
7           1993  TV-MA    125 min   
8           2021  TV-14  9 Seasons   
9           2021  PG-13    104 min   
12          2021  TV-MA    127 min   
24          1998  TV-14    166 min   

                                           listed_in  
7   Dramas, Independent Movies, International Movies  
8                       British TV Shows, Reality TV  
9                                   Comedies, Dramas  
12                      Dramas, International Movies  
24   Comedies, International Movies, Romantic Movies

In [23]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5332 entries, 7 to 8806
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   type          5332 non-null   object
 1   title         5332 non-null   object
 2   director      5332 non-null   object
 3   country       5332 non-null   object
 4   date_added    5332 non-null   object
 5   release_year  5332 non-null   int64 
 6   rating        5332 non-null   object
 7   duration      5332 non-null   object
 8   listed_in     5332 non-null   object
dtypes: int64(1), object(8)
memory usage: 416.6+ KB


In [24]:
df.type.value_counts()

Movie      5185
TV Show     147
Name: type, dtype: int64

**There are 5185 movies and 147 TV shows remaining after cleaning the dataset.**

In [25]:
df.director.value_counts()

Raúl Campos, Jan Suter           18
Marcus Raboy                     15
Jay Karas                        14
Cathy Garcia-Molina              13
Martin Scorsese                  12
                                 ..
Igor Kovalyov, Norton Virgien     1
Danny Cannon                      1
Mana Yasuda                       1
Gupse Özay                        1
Mozez Singh                       1
Name: director, Length: 3945, dtype: int64

In [26]:
df[df["director"] == "Raúl Campos, Jan Suter"]

type                                        title  \
4749  Movie          Coco y Raulito: Carrusel de ternura   
4781  Movie                Luciano Mellera: Infantiloide   
4808  Movie  Jani Dueñas: Grandes fracasos de ayer y hoy   
4849  Movie                     Fernando Sanjiao: Hombre   
4866  Movie                  Carlos Ballarta: Furia Ñera   
4940  Movie            Todo lo que sería Lucas Lauriente   
4962  Movie      Sofía Niño de Rivera: Selección Natural   
5001  Movie            Malena Pichot: Estupidez compleja   
5002  Movie             Natalia Valdebenito: El especial   
5062  Movie                  Sebastián Marcelo Wainraich   
5066  Movie               Ricardo Quevedo: Hay gente así   
5076  Movie               Arango y Sanint: Ríase el show   
5176  Movie                                    Mea Culpa   
5342  Movie               Alan Saldaña: Mi vida de pobre   
5614  Movie                        Daniel Sosa: Sosafado   
5678  Movie           Ricardo O'Farrill: Abrazo navideño   
5839  Movie                Sofía Niño de Rivera: Exposed   
5872  Movie              Ricardo O'Farrill Abrazo Genial   

                    director    country         date_added  release_year  \
4749  Raúl Campos, Jan Suter     Mexico      July 27, 2018          2018   
4781  Raúl Campos, Jan Suter  Argentina       July 6, 2018          2018   
4808  Raúl Campos, Jan Suter      Chile      June 29, 2018          2018   
4849  Raúl Campos, Jan Suter  Argentina       May 25, 2018          2018   
4866  Raúl Campos, Jan Suter     Mexico       May 11, 2018          2018   
4940  Raúl Campos, Jan Suter  Argentina      April 6, 2018          2018   
4962  Raúl Campos, Jan Suter     Mexico     March 30, 2018          2018   
5001  Raúl Campos, Jan Suter  Argentina      March 2, 2018          2018   
5002  Raúl Campos, Jan Suter      Chile      March 2, 2018          2018   
5062  Raúl Campos, Jan Suter  Argentina   January 26, 2018          2018   
5066  Raúl Campos, Jan Suter   Colombia   January 24, 2018          2018   
5076  Raúl Campos, Jan Suter   Colombia   January 17, 2018          2018   
5176  Raúl Campos, Jan Suter     Mexico  November 10, 2017          2017   
5342  Raúl Campos, Jan Suter     Mexico     August 4, 2017          2017   
5614  Raúl Campos, Jan Suter     Mexico   February 3, 2017          2017   
5678  Raúl Campos, Jan Suter     Mexico  December 12, 2016          2016   
5839  Raúl Campos, Jan Suter     Mexico      June 24, 2016          2016   
5872  Raúl Campos, Jan Suter     Mexico   January 22, 2016          2016   

     rating duration        listed_in  
4749  TV-MA   57 min  Stand-Up Comedy  
4781  TV-14   66 min  Stand-Up Comedy  
4808  TV-MA   77 min  Stand-Up Comedy  
4849  TV-MA   63 min  Stand-Up Comedy  
4866  TV-MA   63 min  Stand-Up Comedy  
4940  TV-MA   78 min  Stand-Up Comedy  
4962  TV-MA   62 min  Stand-Up Comedy  
5001  TV-MA   51 min  Stand-Up Comedy  
5002  TV-MA   69 min  Stand-Up Comedy  
5062  TV-MA   49 min  Stand-Up Comedy  
5066  TV-MA   59 min  Stand-Up Comedy  
5076  TV-MA   62 min  Stand-Up Comedy  
5176  TV-MA   54 min  Stand-Up Comedy  
5342  TV-MA   54 min  Stand-Up Comedy  
5614  TV-MA   78 min  Stand-Up Comedy  
5678  TV-MA   31 min  Stand-Up Comedy  
5839  TV-MA   80 min  Stand-Up Comedy  
5872  TV-MA   92 min  Stand-Up Comedy

**There are 18 movies by the pair of directors, Raul Campos and Jan Suter in the dataset, which is the highest.**

In [27]:
df.release_year.value_counts()

2017    657
2018    648
2016    577
2019    519
2020    442
       ... 
1946      1
1961      1
1942      1
1947      1
1944      1
Name: release_year, Length: 72, dtype: int64

**Most of the movies and TV shows in the dataset are released in the year 2017.**

In [28]:
df.listed_in.value_counts()

Dramas, International Movies                               336
Stand-Up Comedy                                            286
Comedies, Dramas, International Movies                     257
Dramas, Independent Movies, International Movies           243
Children & Family Movies, Comedies                         179
                                                          ... 
Comedies, Documentaries                                      1
International TV Shows, Romantic TV Shows, TV Mysteries      1
Horror Movies, International Movies, Sci-Fi & Fantasy        1
Reality TV                                                   1
Cult Movies, Dramas, Thrillers                               1
Name: listed_in, Length: 335, dtype: int64

**There are 335 different genres the movies and shows are listed in but it is evident that genres are basically increased due to Independent Movies.**

In [29]:
df["Genre"] = df.listed_in.apply(lambda x: x if x.find(",") == -1 else x[:x.find(",")])

In [30]:
df.head()

type                          title             director  \
7     Movie                        Sankofa         Haile Gerima   
8   TV Show  The Great British Baking Show      Andy Devonshire   
9     Movie                   The Starling       Theodore Melfi   
12    Movie                   Je Suis Karl  Christian Schwochow   
24    Movie                          Jeans           S. Shankar   

                                              country          date_added  \
7   United States, Ghana, Burkina Faso, United Kin...  September 24, 2021   
8                                      United Kingdom  September 24, 2021   
9                                       United States  September 24, 2021   
12                            Germany, Czech Republic  September 23, 2021   
24                                              India  September 21, 2021   

    release_year rating   duration  \
7           1993  TV-MA    125 min   
8           2021  TV-14  9 Seasons   
9           2021  PG-13    104 min   
12          2021  TV-MA    127 min   
24          1998  TV-14    166 min   

                                           listed_in             Genre  
7   Dramas, Independent Movies, International Movies            Dramas  
8                       British TV Shows, Reality TV  British TV Shows  
9                                   Comedies, Dramas          Comedies  
12                      Dramas, International Movies            Dramas  
24   Comedies, International Movies, Romantic Movies          Comedies

In [31]:
df.Genre.value_counts()

Dramas                          1518
Comedies                        1127
Action & Adventure               806
Children & Family Movies         469
Documentaries                    372
Stand-Up Comedy                  286
Horror Movies                    261
International Movies             109
Classic Movies                    73
Thrillers                         62
International TV Shows            45
Crime TV Shows                    32
British TV Shows                  21
Independent Movies                20
Movies                            20
Anime Features                    18
Music & Musicals                  15
Sci-Fi & Fantasy                  13
Cult Movies                       12
Anime Series                      10
Kids' TV                          10
Docuseries                         7
Stand-Up Comedy & Talk Shows       7
TV Shows                           5
TV Comedies                        3
Romantic Movies                    3
TV Action & Adventure              2
C

**Most TV shows and movies fall in the "Dramas" genre.**

In [32]:
df.rating.value_counts()

TV-MA       1822
TV-14       1214
R            778
PG-13        470
TV-PG        431
PG           275
TV-G          84
TV-Y7         76
TV-Y          76
NR            58
G             40
TV-Y7-FV       3
UR             3
NC-17          2
Name: rating, dtype: int64

**Most shows and movies are rated "TV-MA" i.e. rated for mature audiences.**

In [33]:
df[(df["country"] == "India")]

type                  title                 director country  \
24    Movie                  Jeans               S. Shankar   India   
105   Movie       Angamaly Diaries     Lijo Jose Pellissery   India   
114   Movie                 Anjaam             Rahul Rawail   India   
116   Movie                 Dhanak          Nagesh Kukunoor   India   
118   Movie                Gurgaon            Shanker Raman   India   
...     ...                    ...                      ...     ...   
8772  Movie  Yamla Pagla Deewana 2           Sangeeth Sivan   India   
8773  Movie    Yanda Kartavya Aahe             Kedar Shinde   India   
8798  Movie               Zed Plus  Chandra Prakash Dwivedi   India   
8799  Movie                  Zenda           Avadhoot Gupte   India   
8806  Movie                 Zubaan              Mozez Singh   India   

              date_added  release_year rating duration  \
24    September 21, 2021          1998  TV-14  166 min   
105    September 5, 2021          2017  TV-14  128 min   
114    September 2, 2021          1994  TV-14  143 min   
116    September 2, 2021          2015  TV-PG  114 min   
118    September 2, 2021          2017  TV-14  106 min   
...                  ...           ...    ...      ...   
8772         May 1, 2017          2013  TV-14  147 min   
8773     January 1, 2018          2006  TV-PG  151 min   
8798   December 31, 2019          2014  TV-MA  131 min   
8799   February 15, 2018          2009  TV-14  120 min   
8806       March 2, 2019          2015  TV-14  111 min   

                                              listed_in               Genre  
24      Comedies, International Movies, Romantic Movies            Comedies  
105                Action & Adventure, Comedies, Dramas  Action & Adventure  
114             Dramas, International Movies, Thrillers              Dramas  
116                Comedies, Dramas, Independent Movies            Comedies  
118             Dramas, International Movies, Thrillers              Dramas  
...                                                 ...                 ...  
8772  Action & Adventure, Comedies, International Mo...  Action & Adventure  
8773             Comedies, Dramas, International Movies            Comedies  
8798             Comedies, Dramas, International Movies            Comedies  
8799                       Dramas, International Movies              Dramas  
8806     Dramas, International Movies, Music & Musicals              Dramas  

[875 rows x 10 columns]